### ISSUE TITLE
 False positive W0108 (unnecessary lambda) due to .format()

### ISSUE THREAD

**Issue by scjody** [PyCQA/pylint#2803]
### Steps to reproduce
1. Create `necessary-lambda.py`:

```
def myfunc(f):
    print(f(1))

myfunc(lambda n: "n is: {}".format(n))
```
2. `pylint necessary-lambda.py`

### Current behavior

Messages printed include:

```
necessary-lambda.py:4:7: W0108: Lambda may not be necessary (unnecessary-lambda)
```

### Expected behavior

The warning is not printed

### Notes

The warning is _not_ printed when `%` is used instead of `format()`:

```
def myfunc(f):
    print(f(1))

myfunc(lambda n: "n is: %s" % n)
```

These are obviously contrived examples. The actual code I'm trying to lint uses [factory-boy](https://factoryboy.readthedocs.io/en/latest/index.html):

```
import factory

from models import db, PlotlyUser


class PlotlyUserFactory(factory.alchemy.SQLAlchemyModelFactory):
    class Meta:
        model = PlotlyUser
        sqlalchemy_session = db.session

    id = factory.Sequence(lambda n: n)
    username = factory.Sequence(lambda n: "._-aA{}".format(n))
```

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.6.5 (default, Apr 12 2018, 12:53:59)
[GCC 4.8.4]
```

**Comment by brycepg** [comment]
I can reproduce, thanks

**Comment by brycepg** [comment]
So I'm thinking that the generated code should be changed to: 

```python
def myfunc(f):
    print(f(1))

myfunc("n is: {}".format)
```

to pass this check.

The format method doesn't appear to be exceptional, although the syntax appears a little strange because of the string literal.

This check appears to be doing exactly what it is intended to do for `str.format`. For string interpolation, the lambda is needed because there is no bound method to pass.

**Comment by PCManticore** [comment]
I believe @brycepg is right, your code can be simplified in the form exemplified https://github.com/PyCQA/pylint/issues/2803#issuecomment-471128946.
